registry/handlers/app.go

```diff
@@ -640,6 +640,7 @@ func (app *App) dispatcher(dispatch dispatchFunc) http.Handler {
 		}
 
 		context := app.context(w, r)
+		defer defaultContextManager.release(context)
```

Comment:
The problem with this approach is that in 1.7+, the address of the request object can change between handlers, making the technique used for the context manager invalid. The other PR is much safer.